source/binder.cpp

```diff
@@ -234,19 +254,32 @@ class BinderASTConsumer : public ASTConsumer
 
 class BinderFrontendAction : public ASTFrontendAction {
 public:
+#if  (LLVM_VERSION_MAJOR < 4)
+    virtual ASTConsumerPtr CreateASTConsumer(CompilerInstance &ci, StringRef file) {
+        return (ASTConsumerPtr)( new BinderASTConsumer(&ci) );
+#endif
+#if  (LLVM_VERSION_MAJOR >= 4)
     virtual std::unique_ptr<clang::ASTConsumer> CreateASTConsumer(CompilerInstance &ci, StringRef file) {
         return std::unique_ptr<ASTConsumer>( new BinderASTConsumer(&ci) );
+#endif
```

Comment:
Why this ifdef is needed? Both branches seems to do exactly the same...

Reply:
ASTConsumerPtr is a raw pointer in some llvm 3.X.